constance/forms.py

```diff
@@ -0,0 +1,133 @@
+    str: STRING_LIKE,
+    datetime: (
+        fields.SplitDateTimeField, {'widget': widgets.AdminSplitDateTime}
+    ),
+    date: (fields.DateField, {'widget': widgets.AdminDateWidget}),
+    time: (fields.TimeField, {'widget': widgets.AdminTimeWidget}),
+    float: (fields.FloatField, {'widget': NUMERIC_WIDGET}),
+}
+
+
+def parse_additional_fields(fields):
+    for key in fields:
+        field = list(fields[key])
+
+        if len(field) == 1:
+            field.append({})
+
+        field[0] = import_string(field[0])
+
+        if 'widget' in field[1]:
+            klass = import_string(field[1]['widget'])
+            field[1]['widget'] = klass(
+                **(field[1].get('widget_kwargs', {}) or {})
+            )
+
+            if 'widget_kwargs' in field[1]:
+                del field[1]['widget_kwargs']
+
+        fields[key] = field
+
+    return fields
+
+
+FIELDS.update(parse_additional_fields(settings.ADDITIONAL_FIELDS))
+
+
+if not six.PY3:
+    FIELDS.update({
+        long: INTEGER_LIKE,
+        unicode: STRING_LIKE,
+    })
+
+
+class ConstanceForm(forms.Form):
+    version = forms.CharField(widget=forms.HiddenInput)
+
+    def __init__(self, initial, *args, **kwargs):
+        super(ConstanceForm, self).__init__(*args, initial=initial, **kwargs)
+        version_hash = hashlib.md5()
+
+        for name, options in settings.CONFIG.items():
+            default, help_text = options[0], options[1]
+            if len(options) == 3:
+                config_type = options[2]
+            else:
+                config_type = type(default)
+
+            if config_type not in FIELDS:
+                raise ImproperlyConfigured(_("Constance doesn't support "
+                                             "config values of the type "
+                                             "%(config_type)s. Please fix "
+                                             "the value of '%(name)s'.")
+                                           % {'config_type': config_type,
+                                              'name': name})
+            field_class, kwargs = FIELDS[config_type]
+            self.fields[name] = field_class(label=name, **kwargs)
+
+            version_hash.update(smart_bytes(initial.get(name, '')))
+        self.initial['version'] = version_hash.hexdigest()
+
+    def save(self):
+        for name in settings.CONFIG:
+            setattr(config, name, self.cleaned_data[name])
+
+    def clean_version(self):
+        value = self.cleaned_data['version']
+
+        if settings.IGNORE_ADMIN_VERSION_CHECK:
+            return value
+
+        if value != self.initial['version']:
+            raise forms.ValidationError(_('The settings have been modified '
+                                          'by someone else. Please reload the '
+                                          'form and resubmit your changes.'))
+        return value
+
+
+def get_constance_field(name):
```

Comment:
No need to add an extra function for this, please don't add unnecessary APIs to the app